src/components/Microsoft.DotNet.UpgradeAssistant.Analysis/AnalyzePackageStatus.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.DotNet.UpgradeAssistant.Dependencies;
+using Microsoft.DotNet.UpgradeAssistant.Steps.Packages;
+using Microsoft.DotNet.UpgradeAssistant.Steps.Solution;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Analysis
+{
+    public class AnalyzePackageStatus : IAnalyzeResultProvider
+    {
+        private readonly IDependencyAnalyzerRunner _packageAnalyzer;
+        private readonly IEntrypointResolver _entrypointResolver;
+        private readonly UpgradeOptions _options;
+
+        private IDependencyAnalysisState? _analysisState;
+
+        protected ILogger Logger { get; }
+
+        public AnalyzePackageStatus(UpgradeOptions options,
+            IDependencyAnalyzerRunner packageAnalyzer,
+            ILogger<AnalyzePackageStatus> logger,
+            IEntrypointResolver entrypointResolver)
+        {
+            Logger = logger;
+            _options = options;
+            _packageAnalyzer = packageAnalyzer ?? throw new ArgumentNullException(nameof(packageAnalyzer));
+            _entrypointResolver = entrypointResolver ?? throw new ArgumentNullException(nameof(entrypointResolver));
+            _analysisState = null;
+        }
+
+        public async Task AnalyzeAsync(AnalyzeContext analysis, CancellationToken token)
+        {
+            if (analysis is null)
+            {
+                throw new ArgumentNullException(nameof(analysis));
+            }
+
+            var context = analysis.UpgradeContext;
+            var projects = context.Projects.ToList();
+
+            if (!context.InputIsSolution)
+            {
+                var selectedProject = projects.First(i => i.FileInfo.Name.Equals(Path.GetFileName(context.InputPath), StringComparison.OrdinalIgnoreCase));
```

Comment:
maybe this is earlier in the workflow but I thought the upgrade context already provides access tot he currentProject

Reply:
This is a separate pathway that doesn't use the steps, so that doesn't get set. Maybe something to look into centralizing

Reply:
I removed this part of code to support analysis of all projects.